merge_dynamic_streams_block.py

```diff
@@ -0,0 +1,78 @@
+from collections import defaultdict
+from nio.block.base import Block
+from nio.util.discovery import discoverable
+from nio.signal.base import Signal
+from nio.properties import VersionProperty, TimeDeltaProperty, BoolProperty, \
+    Property
+from nio.modules.scheduler import Job
+
+from nio.block.mixins.group_by.group_by import GroupBy
+
+
+@discoverable
+class MergeDynamicStreams(GroupBy, Block):
```

Comment:
Can we make this the parent class for `MergeStreams`? It feels like `MergeStreams` should be able to use this dynamic functionality and then just specify input_1 and input_2